prow/gcs/reporter/reporter_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/testgrid/metadata"
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+var (
+	fakeGCSServer *fakestorage.Server
+)
+
+type fca struct {
+	c config.Config
+}
+
+func (ca fca) Config() *config.Config {
+	return &ca.c
+}
+
+func sampleProwjob() prowv1.ProwJob {
+	return prowv1.ProwJob{
+		Spec: prowv1.ProwJobSpec{
+			Type: prowv1.PresubmitJob,
+			Refs: &prowv1.Refs{
+				Org:   "kubernetes",
+				Repo:  "test-infra",
+				Pulls: []prowv1.Pull{{Number: 12345}},
+			},
+			Agent: prowv1.KubernetesAgent,
+			Job:   "my-little-job",
+		},
+		Status: prowv1.ProwJobStatus{
+			State:     prowv1.TriggeredState,
+			StartTime: metav1.Time{Time: time.Date(2010, 10, 10, 18, 30, 0, 0, time.UTC)},
+			PodName:   "some-pod",
+			BuildID:   "123",
+		},
+	}
+}
+
+func sampleConfig() config.Getter {
+	return fca{c: config.Config{
+		ProwConfig: config.ProwConfig{
+			Plank: config.Plank{
+				DefaultDecorationConfigs: map[string]*prowv1.DecorationConfig{"*": {
+					GCSConfiguration: &prowv1.GCSConfiguration{
+						Bucket:       "kubernetes-jenkins",
+						PathPrefix:   "some-prefix",
+						PathStrategy: prowv1.PathStrategyLegacy,
+						DefaultOrg:   "kubernetes",
+						DefaultRepo:  "kubernetes",
+					},
+				}},
+			},
+		},
+	}}.Config
+}
+
+func TestMain(m *testing.M) {
```

Comment:
What's the (eventual?) testing plan here? This PR seems to skip any unit testing in favor of two medium tests.

Glancing through maybe these deserve templated unit tests?
`getJobDestination`
`isErrUnexpected`
`reportJobState`

Reply:
All three of those now have templated unit tests.